test/test_task_context.rb

```diff
@@ -2336,6 +2333,16 @@ def assert_process_events_does_not_block
                 end
             end
 
+            it "does not perform setup on configuration" do
+                task = deployment.task("test")
+                flexmock(task).should_receive(:perform_setup).once
```

Comment:
Mocking a method makes it not being called *at all*. I.e. this mock ensures that the next one is never called.

You have to add `.pass_thru` to let the call go through.